Adding submodule neutron-rich-bmm to examples folder

asemposki opened the pull request:
New PR to amend the submodule location to 'examples/neutron-rich-bmm'. 

Task list as before:
- [x] Add SDK and update
- [x] Check and clean up repo 
- [x] Add correct data files
- [x] Check that the README instructions work
- [x] Arrange docs website (e.g., fix docstrings and make sure API is all building well)

Optional additions: `pytest` options.

asemposki committed:
added submodule neutron-rich-bmm to examples folder

wildsm commented:
@asemposki I wasn't sure if the unchecked tasks meant that I should wait to review or whether I can proceed. Let me know.

wildsm commented:
In a fresh env, I started to follow instructions at 

```
(BUQEYE) wild@XPS-13-9310:~/repos$ conda install jupyterlab, jupyter notebook
2 channel Terms of Service accepted
Channels:
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - jupyterlab

Current channels:

  - defaults
```

But if I just do `conda install jupyter noteboook`, then jupyterlab is installed

- [x] Check with others to see if they run into this snag; update README accordingly

Note that if I subsequently `$ conda install jupyterlab`
I get 
```
2 channel Terms of Service accepted
Channels:                                                                       
 - defaults                                                                     
Platform: linux-64                                                              
Collecting package metadata (repodata.json): done                               
Solving environment: done                                                       
                                                                                
                                                                                
==> WARNING: A newer version of conda exists. <==                               
    current version: 25.5.1                                                     
    latest version: 25.9.0                                                      
                                                                                
Please update conda by running                                                  
                                                                                
    $ conda update -n base -c defaults conda                                    
                                                                                
                                                                                
                      
# All requested packages already installed.
```

Proceeding, pip installing gsum resulted in some DEPRECATION warnings, but no errors.

However, pip installing gptools failed:
```
(BUQEYE) wild@XPS-13-9310:~/repos/neutron-rich-bmm/gptools$ pip install .
Processing /home/wild/oldcomp/repos/neutron-rich-bmm/gptools
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 35, in <module>
        File "/home/wild/oldcomp/repos/neutron-rich-bmm/gptools/setup.py", line 9, in <module>
          from Cython.Distutils import build_ext
      ModuleNotFoundError: No module named 'Cython'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

Maybe a missing dependency?

asemposki commented:
@wildsm I knew some things may not work correctly, since I hadn't tested this in a while and it's had issues with some installations before, but testing it now is a good idea so thanks for going ahead and doing that. I will work on the dependencies, I figured some of them may be deprecated or out of sync.

asemposki committed:
Merge pull request #1 from bandframework/develop

Bring up to date with develop

asemposki commented:
Regarding `gptools`, yes, it was missing the Cython dependency. I added it to a requirements.txt file, updated the README to have everyone pip install those requirements, and now `gptools` successfully clones and builds for me.

asemposki committed:
adding reade information

wildsm commented:
Great, that worked. 

I tried a few semi random notebooks.
Neutron_Star_Properties.ipynb - Success
pQCD_EOS.ipynb - Success (some NaN warnings encountered but went to the very end)
NSM_EOS_ChEFT.ipynb - Success (triangle DEPRECATION warning encountered but went to the very end)
Chiral_Cutoff_Comparisons.ipynb - Error: `ModuleNotFoundError: No module named 'arviz'`
BMM_GP_NSM.ipynb - Success

wildsm reviewed (changes requested):
README: 
Between Steps 4 & 5, tell people to navigate to the repository's folder (this is important because the rest of the process happens mostly there, so need to establish what you mean). Even better would be to specify "neutron-rich-bmm folder" in each relevant step so that there is not confusion about whether a dependency repository is being referenced.

Step 9: include the git clone code block for consistency with previous steps

wildsm reviewed (changes requested):
I would relabel 
https://github.com/asemposki/neutron-rich-bmm/blob/main/neutron-rich-bmm_sdkrules.md
to
https://github.com/asemposki/neutron-rich-bmm/blob/main/neutron-rich-bmm_bandsdk.md
to be consistent. 

SDK:
M3 says "The point of contact provided is at as727414@ohio.edu, and is listed in the README.md file." but note that this email does not appear in the README. Isuggest adding the email or just listing the first author's name in M3 (since the email appears at top of this SDK file).

I assume author list in README is intentional, but double check that the additional contributor should not be listed

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

wildsm committed:
Update README.md

wildsm committed:
Update .wordlist.txt

wildsm committed:
Update README.md

Adding neutron-rich-bmm example

wildsm commented:
@asemposki 
I started to add the rest of the places in the bandframework repo where the example will appear.
You can see these at 
https://github.com/bandframework/bandframework/pull/165/files

After the above changes are made, I suspect you might retag a release. In this case, the versions and links will need to be updated in these changed files. (README, exampless/README, SDK list, CHANGELOG, submodule).

Looks great

asemposki commented:
Great. I added the clarifications for the README file. I also added the author names and emails correctly now. I know there are still some data files that need to be added for some notebooks to run correctly, too, so I will do that ASAP.

wildsm commented:
@asemposki I believe this is pending your changes wrt the above reviews, correct?

asemposki commented:
@wildsm Yes, I've been working on fixing things, and so far I've got the API building for the docs website, and I'm working on making sure I include the proper data files for the notebooks that are publicly available to use. I should be done with everything in the next couple of days at the most.

asemposki committed:
updating submodule SAMBA

asemposki commented:
Well, I guess SAMBA is also going to be updated on this PR. All that was changed there was a fix to the CI workflow and a fix to the rendering of the API documentation.

wildsm committed:
Merge branch 'develop' into develop

asemposki committed:
updating neutron-rich-bmm with data and SDK renaming

asemposki commented:
OK the CI build is now passing, and I still need to make some docstring changes, but all fixes should be now included.

asemposki committed:
updating submodule and new release of neutron-rich-bmm

asemposki commented:
OK all my changes are finished other than the release updating. I will do that now and then request a review again.

asemposki committed:
Update neutron-rich-bmm version in CHANGELOG

Updated the version of neutron-rich-bmm from v0.0.1 to v0.1.0 in the changelog.

asemposki committed:
updating version of submodule neutron-rich-bmm

asemposki committed:
updating the neutron-rich-bmm link in README

asemposki reviewed (commented):
OK I've updated all of the files where the release will be mentioned/linked, from what I can tell.

wildsm reviewed (commented):
Evolving review:

- [x] Done:
> I would relabel
> https://github.com/asemposki/neutron-rich-bmm/blob/main/neutron-rich-bmm_sdkrules.md
> to
> https://github.com/asemposki/neutron-rich-bmm/blob/main/neutron-rich-bmm_bandsdk.md
> to be consistent

- [x] Done:
> SDK:
> M3 says "The point of contact provided is at [as727414@ohio.edu](mailto:as727414@ohio.edu), and is listed in the README.md file." 
> but note that this email does not appear in the README. I
> suggest adding the email or just listing the first author's name in M3 (since 
> the email appears at top of this SDK file).

- [x] Done:
> README:
> Between Steps 4 & 5, tell people to navigate to the repository's 
> folder (this is important because the rest of the process happens 
> mostly there, so need to establish what you mean). Even better 
> would be to specify "neutron-rich-bmm folder" in each relevant 
> step so that there is not confusion about whether a dependency 
> repository is being referenced.
> 
> Step 9: include the git clone code block for consistency with previous steps

wildsm commented:
@asemposki The new SAMBA submodule that is pointed to here is
https://github.com/asemposki/SAMBA/tree/fbdbfaeb1b3bd39c8aca484588ac7bcf82ac146d
 which is v1.2.0. Do you want to keep that as (with the README citing v1.1.0, see  https://github.com/asemposki/SAMBA/issues/5 )?

wildsm reviewed (approved):
Modulu the note above (which may result in a slight retagging of SAMBA and hence trickle down to: the submodule, the changelog, the primary readme, the example readme), this looks great. I'm approving so that if the response is to not change the SAMBA readme, we're good to merge.

asemposki committed:
updating submodule SAMBA for readme changes

asemposki committed:
trying to merge so we can update

asemposki commented:
I have made the SAMBA README changes which required me to update submodule for that commit and I had to fix a divergent branch with my local copy of bandframework's develop branch. These should be good to go but I'm going to hit request review again now so you can approve @wildsm.

wildsm reviewed (approved):
Looks great. Updates SAMBA readme to be inline with referenced version.
